src/cmake/thirdparty/SetupGeosxThirdParty.cmake

```diff
@@ -612,19 +626,26 @@ if(DEFINED HYPRE_DIR AND ENABLE_HYPRE)
                       EXTRA_LIBRARIES ${EXTRA_LIBS}
                       DEPENDS ${HYPRE_DEPENDS})
 
+    extract_version_from_header( NAME hypre
+                                 HEADER "${HYPRE_DIR}/include/HYPRE_config.h"
+                                 VERSION_STRING "HYPRE_RELEASE_VERSION" )
+
+    # Extract some additional information about development version of hypre
+    file(READ ${HYPRE_DIR}/include/HYPRE_config.h header_file)
+    if("${header_file}" MATCHES "HYPRE_DEVELOP_STRING *\"([^\"]*)\"")
+        set(hypre_dev_string "${CMAKE_MATCH_1}")
+        if("${header_file}" MATCHES "HYPRE_DEVELOP_BRANCH *\"([^\"]*)\"")
+            set(hypre_dev_branch "${CMAKE_MATCH_1}")
+        endif()
+        set(hypre_VERSION "${hypre_dev_string} (${hypre_dev_branch})" CACHE STRING "" FORCE)
```

Comment:
I find it useful to know the name of the branch associated with the commit hash. `HYPRE_BRANCH_NAME` provides that info. Should it be added here? PS: very nice organization work

Reply:
@victorapm I think I just took the macros that we used to print out in `HypreInterface::initialize()` and moved them to CMake-time for consistency with other dependecies. `HYPRE_DEVELOP_STRING` and `HYPRE_DEVELOP_BRANCH` are the ones we used to print. Should I replace `HYPRE_DEVELOP_BRANCH` with `HYPRE_BRANCH_NAME`? What is the difference between the two?

Reply:
Good! `HYPRE_BRANCH_NAME` gives the name of the branch that a particular commit belongs to, `HYPRE_DEVELOP_BRANCH` gives the name of the main development branch in hypre (master). So the first is more general and should be the preferable choice.

Lastly, I think you are aware but just mentioning here: these macros are only defined when building hypre from  directory that has been cloned from git, so they don't exist for hypre tarballs downloaded from git or releases